# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/CommonSocketControl.h
# Commit: 48ce2b670f32
# Full Hash: 48ce2b670f32691b0537c13c165ec8459d74907c
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-09-25 09:36:42
# Regressor Bug: 1577643
# File Overlap Count: 1
# Description:
#   Bug 1577643 - Implement a security info class for the quic transport. r=keeler
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44073
# ==============================================================================

diff -r 2a39731bf265 -r 48ce2b670f32 security/manager/ssl/CommonSocketControl.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/security/manager/ssl/CommonSocketControl.h	Tue Sep 24 20:56:39 2019 +0000
@@ -0,0 +1,40 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef CommonSocketControl_h
+#define CommonSocketControl_h
+
+#include "nsISSLSocketControl.h"
+#include "TransportSecurityInfo.h"
+
+class CommonSocketControl : public mozilla::psm::TransportSecurityInfo,
+                            public nsISSLSocketControl {
+ public:
+
+
+  NS_DECL_ISUPPORTS_INHERITED
+  NS_DECL_NSISSLSOCKETCONTROL
+
+  explicit CommonSocketControl(uint32_t providerFlags);
+
+  uint32_t GetProviderFlags() const { return mProviderFlags; }
+  void SetSSLVersionUsed(int16_t version) { mSSLVersionUsed = version; }
+  void SetResumed(bool aResumed) { mResumed = aResumed; }
+
+ protected:
+  ~CommonSocketControl() = default;
+  nsCString mNegotiatedNPN;
+  bool mNPNCompleted;
+  bool mHandshakeCompleted;
+  bool mJoined;
+  bool mSentClientCert;
+  bool mFailedVerification;
+  mozilla::Atomic<bool, mozilla::Relaxed> mResumed;
+  uint16_t mSSLVersionUsed;
+  uint32_t mProviderFlags;
+};
+
+#endif  // CommonSocketControl_h